packages/components/organisms/f-form/src/components/Form.vue

```diff
@@ -84,20 +129,62 @@ export default {
     },
 
     methods: {
+        isFormValid () {
+            this.$v.$touch();
+            return !this.$v.$invalid;
+        },
+
         updateField ({ fieldName, value }) {
-            return this.$emit(FORM_EVENTS.fieldUpdated, { fieldName, value });
+            this.$emit(FORM_EVENTS.fieldUpdated, { fieldName, value });
         },
 
         onFormSubmit () {
             this.$emit(FORM_EVENTS.submitting);
+
+            if (this.isFormValid()) {
+                this.$emit(FORM_EVENTS.valid, this.formFields);
+            } else {
+                const validationState = validations.getFormValidationState(this.$v);
+                this.scrollToFirstInlineError();
+                this.$emit(FORM_EVENTS.invalid, validationState);
+            }
+        },
+
+        isValidPhoneNumber () {
+            const phoneNumberValue = this.formData.formFields.find(field => field.name === 'mobileNumber');
+
+            return phoneNumberValue && validations.isValidPhoneNumber(phoneNumberValue.value, this.locale);
         },
 
-        fieldProps (field) {
-            return {
-                name: field.name,
-                value: field.value || '',
-                'label-text': field.translations?.label
-            };
+        isValidPostcode () {
+            const postcodeValue = this.formData.formFields.find(field => field.name === 'postcode');
+
+            return postcodeValue && validations.isValidPostcode(postcodeValue.value, this.locale);
+        },
+
+        translations (field) {
+            const formField = this.formData.formFields.find(fieldData => fieldData.name === field);
+
+            return formField?.translations;
+        },
+
+        hasValidationMessages (field) {
+            return !!this.translations(field)?.validationMessages;
+        },
+
+        hasInvalidErrorMessage (field) {
+            return !!this.translations(field)?.validationMessages?.invalid;
+        },
+
+        scrollToFirstInlineError () {
+            this.$nextTick(() => {
+                const scrollingDurationInMilliseconds = 650;
```

Comment:
Out of interest how was `650` selected?

Reply:
I'm not sure but I'll look into it. This has been copied over directly from checkout but I can see if I can find out and add a note

Reply:
Struggling to find any information of why that number was initially picked when checkout was built. As soon as I find out I'll add a comment in